### ISSUE TITLE
Metadata string value "..." contains illegal characters

### ISSUE THREAD

**Issue by samarthagarwal** [firebase/firebase-functions#463]
### Version info

**node:** 10.13.0

**firebase-functions:** ^2.3.1

**firebase-tools:** 6.11.0

**firebase-admin:** 8.1.0

### Test case

Inside a cloud function (HTTP or Firestore trigger), if we try to read a document nested within other documents, special characters in the document name are cause the function to crash.

So the following code crashes if `churchId` is `Sørlandskirken` or something like that. I tested and it allows spaces but not special characters like `ø` and `æ`. I have documents in my Cloud Firestore database with names with these characters and everything else works fine with the client SDK.

```
let programItems: QuerySnapshot = await admin.firestore().collection("Denominations/NO-Pinsebevegelsen/Churches/" + churchId + "/ProgramItems").get();
```
I get the error message that says.

```
Metadata string value "parent=projects/jesuskidsflutter-prod/databases/(default)/documents/Denominations/NO-Pinsebevegelsen/Churches/Sørlandskirken" contains illegal characters
```

### Steps to reproduce

Create a document inside any collection with document ID as a string that contains special characters like `ø` and `æ`.
Create a sub-collection within this collection and create some documents inside the subcollections.
Next, try to read the documents such that the path to the document being read contains the document ID with special characters using the NodeJS SDK.

### Expected behavior

The documents should be read without error. It handles spaces well but not special characters like `ø` and `æ`. I have not tested with other special characters.

### Actual behavior

I get the error message that says.
```
Metadata string value "parent=projects/jesuskidsflutter-prod/databases/(default)/documents/Denominations/NO-Pinsebevegelsen/Churches/sørlandskirken" contains illegal characters
```
### Were you able to successfully deploy your functions?

Yes, but function crashes because of this error. Please note that I am successfully able to create a Firestore trigger for the same path (that contains the special characters) and it works. Just can't read the documents.

**Comment by kevinajian** [comment]
Hi @samarthagarwal,

Sorry you're running into this issue. I was able to reproduce this on firebase-admin 8.1.0 and will look into this. In the meantime, I was able to get it working by downgrading to firebase-admin 7.0.0, so you can try using that if possible.

**Comment by kevinajian** [comment]
Repro with `admin.firestore().collection('test/testøæ/programItems').get();`

```
Error: Metadata string value "parent=projects/kjian-19-6-10/databases/(default)/documents/test/testøæ" contains illegal characters
    at validate (/srv/functions/node_modules/@grpc/grpc-js/build/src/metadata.js:48:23)
    at Metadata.set (/srv/functions/node_modules/@grpc/grpc-js/build/src/metadata.js:70:9)
    at Object.buildMetadata [as metadataBuilder] (/srv/functions/node_modules/google-gax/build/src/grpc.js:186:38)
    at /srv/functions/node_modules/google-gax/build/src/normalCalls/timeout.js:56:25
    at Object.request (/srv/functions/node_modules/google-gax/build/src/streamingCalls/streaming.js:117:36)
    at makeRequest (/srv/functions/node_modules/retry-request/index.js:125:28)
    at retryRequest (/srv/functions/node_modules/retry-request/index.js:94:5)
    at StreamProxy.setStream (/srv/functions/node_modules/google-gax/build/src/streamingCalls/streaming.js:108:33)
    at StreamingApiCaller.call (/srv/functions/node_modules/google-gax/build/src/streamingCalls/streamingApiCaller.js:68:16)
    at funcPromise.then.then.apiCall (/srv/functions/node_modules/google-gax/build/src/createApiCall.js:84:30)
```

**Comment by samarthagarwal** [comment]
@kevinajian Thanks for looking into this. I actually ended up changing the names of my documents (copying documents with names that do not have such character), but thanks for the info.

**Comment by kevinajian** [comment]
Internal bug 135490750

**Comment by schmidt-sebastian** [comment]
We found the root cause (we are using the wrong encoding in our HTTP headers) and will fix this as part of https://github.com/googleapis/gax-nodejs/issues/520

**Comment by schmidt-sebastian** [comment]
This has been fixed as part of googleapis/gax-nodejs#520. If you re-install your npm modules, you will pull this fix in automatically. Thank you for filing this issue!
